### DISCUSSION TITLE
Issues with Controlling Holybro QAV250 via Crazyflie 2.1 with BigQuad Deck

### DISCUSSION THREAD

**Discussion by Daneto96** [bitcraze/discussions#1370] [Q&A]
Hello everyone,

I am working on a university project at Hochschule München where I am trying to control a larger drone (Holybro QAV250) using a Crazyflie 2.1 with a BigQuad Deck. Here is what I have done so far:

**Custom Firmware:** 
I created a custom firmware via make menuconfig with the "Support BigQuad deck" option enabled.

Motor Configuration:

Enabled "Arming"
Set an idle speed of 5000
Configured the ESC protocol to OneShot125 (also tried with PWM)

**Connection and Initial Testing:**
Connected the Crazyflie with the Crazyflie client.
When pressing the arming button, the QAV250 motors start spinning at idle speed.
I used a gaming controller within the Crazyflie client to test the drone's response.

**Issue:**
For a brief period, the controls worked, but then the message "Locked! Reboot Required" appeared in the client. After rebooting the drone and reconnecting, the motors continue to spin at idle speed, but the drone does not respond to any thrust inputs from the controller, despite the thrust input being displayed correctly in the client.

Are there any recommended configurations or settings to avoid this issue?
Any insights on why the drone does not respond to thrust inputs after rebooting?
Any help or suggestions would be greatly appreciated!

kind regards
Daniel

**Comment by tobbeanton** [comment]
It sounds like there could be a bug involved. What version of the cflib, cfclient and firmware are you running?

> For a brief period, the controls worked, but then the message "Locked! Reboot Required"

Maybe there is a lot of vibrations that is causing it to lock. You could try and increase the limits, start with the [freeFallThreshold](https://github.com/bitcraze/crazyflie-firmware/blob/master/src/modules/src/supervisor.c#L206), like 0.3

**Reply by heinrichwagner** [reply]
Hi @tobbeanton, 

I am also participating in the same project as @Daneto96 .

Thank you for your suggestion. 

Of course, we still have to investigate why the drone changes to the "locked" status, but the bigger problem we are currently having is that after a restart (whether it's just Crazyflie or the whole Holybro and Crazyflie combination), the motors can no longer be controlled via the gamepad. Only the idle thrust can be switched on.

We found a weird workaround: 
In "Battery unplugged" state: If we unplug the BigQuad Deck from the ESCs and plug it back in, plugin the battery, reconnect with cfclient => we can control the motors via the gamepad again

We tried it with:
Firmware: 2024.2
cfclient: 2024.3 and 2023.6
cflib: 0.1.25.1

**Reply by tobbeanton** [reply]
Maybe it is something with the ESCs not "calibrating" and accepting input signal? But you say "Only the idle thrust can be switched on." which indicates that the ESCs works fine and that something else is preventing the gampad inputs from going through?

In the cfclient, do you see a response from the motor measurements?
![image](https://github.com/bitcraze/discussions/assets/6593186/25e91f16-012a-4346-be4c-87511145020b)

**Reply by Daneto96** [reply]
Thanks for your answers!

Yes - We do see a response in the cfclient from the motor measurements.
A thrust percentage is displayed (over 70%) as well as the percentages for M1, M2, M3 and M4 (each 7% corresponding to idle thrust).

**Reply by tobbeanton** [reply]
I tried the same thing and could not reproduce your problem but I might have an idea. Have you activated the arming functionality? Because if the ESC takes to long to boot and idleThrust is set, the startup zero thrust might never get received by the ESC, and will thus not calibrate.

**Reply by heinrichwagner** [reply]
Yes, we activated the arming functionality. 

We also checked the ESC´s with a frequency generator. They worked fine. 
The propeller test also works, when setting the health.propTestPWMRatio and enable health.startPropTest.

Do the crazyflie need some ACK Signal from the ESC´s or motor during start up?

**Reply by tobbeanton** [reply]
> Do the crazyflie need some ACK Signal from the ESC´s or motor during start up?

No. Do you happen to have a scope so you can check what is being sent out to the ESC when the motors are not working?

**Reply by heinrichwagner** [reply]
Hi tobbeanton, 
nothing is being sent to the ESC, when the motors are not working.

We were able to narrow down the problem a little bit. 

We unplugged the crazyflie completely from the bigger drone and used another BigQuad deck. 
We could see the blue bars goes up and down in the Cfclient when triggering the thrust.
![335492712-25e91f16-012a-4346-be4c-87511145020b](https://github.com/bitcraze/discussions/assets/97028723/964e4183-1f24-45ba-9d27-ebec03ad9f98)

After a couple of restarts, the control of the motors stops working. Only the idle motor control is shown.

So the crazyflie with BigQuad deck unplugged from the bigger drone, shows the same failure behaviour as plugged.

Can you recommend a crazyflie firmware version that should run stable with BigQuad Deck?

**Reply by tobbeanton** [reply]
> So the crazyflie with BigQuad deck unplugged from the bigger drone, shows the same failure behaviour as plugged.

To me this sounds like the supervisor is locking it for some reason. What does the supervisor info show at this point?
![image](https://github.com/bitcraze/discussions/assets/6593186/ef81e2ce-d577-45ef-a99c-9355cff18859)

To you have as step by step guide we could follow to replicate the problem?

**Reply by heinrichwagner** [reply]
Hi @tobbeanton, 
sorry for the late answer. 

It looks like the problem is solved, by just adding the propellers to the motors. 
We did the tests before without the propellers.

We added propellers, to do some PID tuning and didn´t run again into the described problem above. 

I assume the crazyflie expects a reaction to the control of the motors and if this doesn't happen, it locks itself?

**Reply by tobbeanton** [reply]
Not actually sure why it locks without the propellers but it sounds like the estimator goes bananas. Glad you found a solution!
